Build failure: pdfpc

### Steps To Reproduce

`nix build nixpkgs#pdfpc`

### Build log

```
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j4 SHELL=/nix/store/j80wcbm6bb1nsqsgmniq43rg980p27q9-bash-5.2-p15/bin/bash
[  2%] Generating classes/action/action_mapping.c, classes/action/link_action.c, classes/action/movie.c, classes/config_file_reader.c, classes/dbus_server.c, classes/drawings/drawing.c, classes/drawings/drawing_commands.c, classes/drawings/storage.c, classes/metadata/pdf.c, classes/metadata/xml.c, classes/options.c, classes/presentation_controller.c, classes/renderer/cache.c, classes/renderer/image.c, classes/renderer/markdown.c, classes/renderer/pdf.c, classes/rest_server.c, classes/timer.c, classes/view/behaviour/base.c, classes/view/behaviour/pdf_link.c, classes/view/markdown.c, classes/view/pdf.c, classes/view/transition_manager.c, classes/view/video.c, classes/window/fullscreen.c, classes/window/overview.c, classes/window/presentation.c, classes/window/presenter.c, classes/window/qrcode.c, interfaces/controllable.c, pdfpc.c
/build/source/src/classes/drawings/drawing_commands.vala:57.46-57.59: error: `pdfpc.DrawingCommand' is not a supported generic type argument, use `?' to box value types
   57 |             this.drawing_commands = new List<DrawingCommand>();
      |                                              ^~~~~~~~~~~~~~
/build/source/src/classes/drawings/drawing_commands.vala:58.43-58.56: error: `pdfpc.DrawingCommand' is not a supported generic type argument, use `?' to box value types
   58 |             this.redo_commands = new List<DrawingCommand>();
      |                                           ^~~~~~~~~~~~~~
/build/source/src/classes/drawings/drawing_commands.vala:73.43-73.56: error: `pdfpc.DrawingCommand' is not a supported generic type argument, use `?' to box value types
   73 |             this.redo_commands = new List<DrawingCommand>(); // clear
      |                                           ^~~~~~~~~~~~~~
Compilation failed: 3 error(s), 0 warning(s)
make[2]: *** [src/CMakeFiles/pdfpc.dir/build.make:108: src/classes/action/action_mapping.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:169: src/CMakeFiles/pdfpc.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

none

### Notify maintainers

@pSub 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.25, NixOS, 23.05 (Stoat), 23.05.20230430.da45bf6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/store/lyaypyw1xhd1h8ianv6siqm3j1wq51cg-source`

```

--- a/pkgs/applications/misc/pdfpc/default.nix
+++ b/pkgs/applications/misc/pdfpc/default.nix
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchFromGitHub, cmake, pkg-config, vala, gtk3, libgee
 , poppler, libpthreadstubs, gstreamer, gst-plugins-base, gst-plugins-good, gst-libav, gobject-introspection, wrapGAppsHook
-, qrencode, webkitgtk, discount, json-glib }:
+, qrencode, webkitgtk, discount, json-glib, fetchpatch }:
 
 stdenv.mkDerivation rec {
   pname = "pdfpc";
@@ -31,6 +31,16 @@ stdenv.mkDerivation rec {
     webkitgtk
     discount
     json-glib
+  ];
+
+  patches = [
+    # needed for compiling pdfpc 4.6.0 with vala 0.56.7, see
+    # https://github.com/pdfpc/pdfpc/issues/686
+    # https://github.com/pdfpc/pdfpc/pull/687
+    (fetchpatch {
+      url = "https://github.com/pdfpc/pdfpc/commit/d38edfac63bec54173b4b31eae5c7fb46cd8f714.diff";
+      hash = "sha256-KC2oyzcwU2fUmxaed8qAsKcePwR5KcXgpVdstJg8KmU=";
+    })
   ];
 
   cmakeFlags = lib.optional stdenv.isDarwin "-DMOVIES=OFF";

